dch/dch.js

```diff
@@ -1,86 +1,67 @@
-// Mongoose. This can be made more elaborate but it is enough to fetch all
-// data required as a single JSON object.
-const acccelerationCollectionDataTypes = {data: String};
-const accelerationCollection = {collection: 'mqtt-data'};
-const watchDataModelName = 'watchDataModel';
-
-// Conosle messages and errors
-const SERVER_INITIALIZED_MESSAGE = 'View page in browser at ' +
-  '"http://localhost:3000"';
-const DB_CONNECTION_ESTABLISHED_MESSAGE = 'Connection made with MongoDB';
-
-const TOTAL_NODEJS_MQTT_CLIENTS = 2;
-
-// Database schema and connection configurations
-let watchDataSchema = mongoose.Schema(
-  acccelerationCollectionDataTypes,
-  accelerationCollection
-);
-
-let watchDataModel = mongoose.model(watchDataModelName, watchDataSchema);
-let db = mongoose.connection;
-
-let app = express();
-let systemListener = mqttSubscriber.sysClient;
+const API_BASE_PATH = '/api';
+const PUBLIC_DIR = 'public';
+//API paths
+const DELETE_DATA_PATH = '/deleteData/:watchId';
+const GET_WATCH_ID = '/getWatchID'
+const TOTAL_CONNECTED_DEVICES = '/total_connected_devices';
+const GET_DATA_PATH = '/get/:watchiId/:StartTime/:stopTime/:frequence';
+
+//make the home directory to be public
+app.use(express.static(PUBLIC_DIR));
+
+//Decrypt the data here
+router.use(function(req, res, next) {
+    console.log('The data we get '+res);
+    next(); // make sure we go to the next routes and don't stop here
+});
 
-// Connect Mongoose to the database
-mongoose.connect(MONGODB_URL);
 
-db.once(MONGOOSE_OPEN, function() {
-  console.log(DB_CONNECTION_ESTABLISHED_MESSAGE);
+router.delete(DELETE_DATA_PATH, function(req, res) {
+	//putd data for deleting data for a watch id
+    	req.json({watchID: req.params.watchId});
 });
 
-// Tells the app to serve DCGUI files in the 'public' directory
-app.use(express.static(PUBLIC_DIRECTORY));
-app.get(NO_PATH, function (request, res) {
-  res.sendfile();
+router.get(GET_DATA_PATH, function(req, res){
+	try{
+		var data = database.getData(req.params.watchId, req.params.StartTime, req.params.stopTime,req.params.frequence );
+		res.json(data);
+	}catch(e){
+		console.log('There was an error getting data from database: ' + e);
+	}
 });
 
-// API functions
-app.get(GET_DATA_PATH, function (request, res) {
-  watchDataModel.find(function (err, data) {
-      return res.end(JSON.stringify(data));
-    });
+router.get(GET_WATCH_ID, function(req, res){
+	console.log('At get watch id');
+	//database.getData(watch_id, startTime, stopTime, freq);
+	
 });
 
-app.get(TOTAL_CONNECTED_DEVICES_PATH, function (request, res) {
-  // Message for debugging
-  console.log('Total watch clients: ' + systemListener.totalClients);
-  var totalClients = parseInt(systemListener.totalClients);
-  var totalWatches = totalClients - TOTAL_NODEJS_MQTT_CLIENTS;
-  return res.end(totalWatches.toString());
+router.get(TOTAL_CONNECTED_DEVICES, function(req, res){
+	console.log('Total watch clients: ' + systemListener.totalClients);
```

Comment:
Agreed. Also a word of warning these new files have broken the part of home page which has shows the number of connected devices. I couldn't find the piece of code which needed to be changed but now all api calls will be begin with /api/.....